Nonprofit-Exchange-Hub/web-app
# Add to swagger API documentation (part 2)

## Dev Summary

Follow-up to [PR#336](https://github.com/Nonprofit-Exchange-Hub/web-app/pull/336).  This PR has updates to `message`, `user-org`, and `transaction` controllers and dtos to expose information regarding API requests and responses.




## Test Plan


repro steps:

1. localhost:3001/docs 
2. look through and verify the details displayed in `message`, `user-org`, and `transaction` endpoints
3.  test login, password reset, and CRUD operations for `assets` and `categories`

When testing requests on Swagger, we'll need to authorize first by copying and pasting `NEH_is_cool` cookie value:
- run `localhost:3000` and login via `http://localhost:3000/login`
- open dev console and go to `Application` tab
- in `Storage` section, click on `Cookies` (should now see `NEH_is_cool`)
<img width="947" alt="Screenshot 2023-05-04 at 11 38 50 AM" src="https://user-images.githubusercontent.com/2562488/236298273-8a71d9c0-be1b-4dea-87f3-df7b2286ff2d.png">
- in `localhost:3001/docs`, click on `Authorize` button and enter the cookie value
<img width="1425" alt="Screenshot 2023-05-04 at 11 40 02 AM" src="https://user-images.githubusercontent.com/2562488/236298535-e323f798-e8a2-4f2d-b335-1a7abdb5c18c.png">


## Resources

- [NextJS Swagger documentation](https://docs.nestjs.com/openapi/introduction)
